### PR TITLE
Database based on diagram

### PR DESCRIPTION
**Pull request by ecekpo**
This project is between Emem Ekpo and Orji Maxwell
--Added updated--
-Joined tables from many-to-many relationships
- FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    FOREIGN KEY(id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoices (
+    id serial primary key NOT NULL,
+    total_amount decimal,
+    generated_at timestamp,
+    payed_at timestamp,
+    medical_history_id int
+    date_of_birth date
```
Comment (Nemwel-Boniface) [inline schema_based_on_diagram.sql:34]: - [ ] Checking through the provided ERD, this field is not supposed to be here.

image

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	  REFERENCES patient(patient_id)
+);
+
+CREATE TABLE treatments (
+    id serial primary key NOT NULL,
+    type varchar,
+    name varchar 
+    CONSTRAINT fk_patient
+    FOREIGN KEY(id) REFERENCES medical_histories(id)
+);
```
Comment (Nemwel-Boniface) [inline schema_based_on_diagram.sql:26]: - [ ] As it is written in the project requirements, you are also required to create all join tables wherever necessary. Kindly ensure that you create the join table. **Hint:** a table between medical_histories and treatments

image

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    id serial primary key NOT NULL,
+    unit_price decimal,
+    quantity int,
+    total_price decimal,
+    invoice_id int,
+    treatment_id int 
+    CONSTRAINT fk_invoice
+    FOREIGN KEY(id) REFERENCES invoice_id(id)
+    FOREIGN KEY(id) REFERENCES treatments(id)
+);
```
Comment (Nemwel-Boniface) [inline schema_based_on_diagram.sql]: - [ ]  As it is written in the project requirements, you are required to write the indexes for all your foreign keys. As you will be having 6 tables, this means that you also need 6 indexes as well for each of the tables. Kindly include them, **Example:** ```CREATE INDEX ON medical_histories (patient_id);```

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @ecekpo  and @Maxwell011     :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (ecekpo) [reply schema_based_on_diagram.sql:34]: Hello Nemwel-Boniface,
Good evening. Please, the above table is there among other tables provided.
Thank you.
Emem Ekpo

**Review by Davidosky007 — APPROVED** [review]
Hi team , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tables created properly 
* Join table created properly 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Nemwel-Boniface) [reply schema_based_on_diagram.sql:34]: Oh, my apologies @ecekpo . I meant the date of birth field. Good job for making the changes

**Pull request merged**